back/app/services/timeline_service.rb

```diff
@@ -79,6 +79,7 @@ def timeline_active(project)
   end
 
   def timeline_active_on_collection(projects)
+    projects = projects.to_a
```

Comment:
This worries me a little bit. A long time ago, I think @kogre also optimized this service (but mainly in addition to what is now the `ParticipationPermissionsService`) so that it would be efficiently using the Rails cache when the same queries are executed multiple times in the same request.

However, a lot has changed since then, so it's possible that this code is no longer as efficient as it used to be.